airbyte-integrations/connectors/source-github/source_github/utils.py

```diff
@@ -40,49 +55,94 @@ class MultipleTokenAuthenticatorWithRateLimiter(AbstractHeaderAuthenticator):
     the first token becomes available again.
     """
 
-    DURATION = 3600  # seconds
+    DURATION = pendulum.duration(seconds=3600)  # Duration at which the current rate limit window resets
 
-    def __init__(self, tokens: List[str], requests_per_hour: int, auth_method: str = "Bearer", auth_header: str = "Authorization"):
+    def __init__(self, tokens: List[str], auth_method: str = "Bearer", auth_header: str = "Authorization"):
         self._auth_method = auth_method
         self._auth_header = auth_header
-        now = time.time()
-        self._requests_per_hour = requests_per_hour
-        self._tokens = {t: SimpleNamespace(count=self._requests_per_hour, update_at=now) for t in tokens}
+        now = pendulum.now()
+        self._tokens = {
+            t: Token(count_rest=5000, count_graphql=5000, update_at=now, reset_at_rest=now, reset_at_graphql=now) for t in tokens
```

Comment:
Can we make a `dict` and map it directly to the Token class like `t: Token(**token_args) for t in tokens?`